### PR TITLE
es-313 Add configuration support for scrape interval in Target Allocator

### PR DESCRIPTION
**Pull request by daidokoro**
This PR allows configuration of scrape interval value for target allocator prometheus custom resources.

- added scrape interval configuration condition
- added scrapeInterval to  schema
- added scrapeInterval default value
- chart version bump
- changelog

### CODE REVIEW

File: charts/opentelemetry-collector/templates/configmap-targetallocator.yaml [hunk 1]
```diff
@@ -16,7 +16,11 @@ data:
         app.kubernetes.io/name: {{ include "opentelemetry-collector.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
     prometheus_cr:
+      {{- if .Values.targetAllocator.prometheusCR.scrapeInterval }}
+      scrape_interval: {{ .Values.targetAllocator.prometheusCR.scrapeInterval }}
+      {{- else }}
       scrape_interval: 30s
```
Comment (nicolastakashi) [inline charts/opentelemetry-collector/templates/configmap-targetallocator.yaml:22]:
```suggestion
      scrape_interval: {{ .Values.targetAllocator.prometheusCR.scrapeInterval | default "30s" }}
```

wdyt?

File: charts/opentelemetry-collector/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
 ## OpenTelemetry Collector
+### v0.89.1 / 2024-07-31
```
[the remark this answers is not in the document]
Comment (daidokoro) [reply charts/opentelemetry-collector/CHANGELOG.md:4]: Thanks, adjusted.

Comment (royfur) [inline charts/opentelemetry-collector/CHANGELOG.md:4]: need to update to 88.6 and the date

Comment (daidokoro) [reply charts/opentelemetry-collector/CHANGELOG.md]: thanks, updated.

**Pull request merged**